mautic/mautic
# Support for emoji in emails

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | N.A.
| Related developer documentation PR URL | N.A.
| Issue(s) addressed                     | N.A.




#### Description:
The issue with email subject is that it is converted to a token and so fails with email_stats.tokens on send because that column is not utf8mb and thus does not support emojis natively.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Execute query `ALTER TABLE <prefix>email_stats CHANGE COLUMN tokens tokens LONGTEXT CHARACTER SET 'utf8' COLLATE 'utf8_unicode_ci' NULL DEFAULT NULL COMMENT '(DC2Type:array)';` to change the email_stats:token column's charset to utf8.
2. Create email with emoji in subject (Use this emoji: https://emojipedia.org/worm/)
2. It is observed that email is not received when sent (use campaign send email action to send emails)

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Revert the query above if it was executed using `ALTER TABLE <prefix>email_stats CHANGE COLUMN tokens tokens LONGTEXT CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_ci' NULL DEFAULT NULL COMMENT '(DC2Type:array)';` to change charset to utf8mb4.
3. Follow steps to reproduce, email should be sent and received.
